infra/src/main/kotlin/fi/fta/geoviite/infra/localization/LocalizationController.kt

```diff
@@ -0,0 +1,31 @@
+package fi.fta.geoviite.infra.localization
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import fi.fta.geoviite.infra.authorization.AUTH_ALL_READ
+import fi.fta.geoviite.infra.logging.apiCall
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+
+@RestController
+@RequestMapping("/localization")
+class LocalizationController @Autowired constructor(
+    val localizationService: LocalizationService,
+) {
+    private val logger: Logger = LoggerFactory.getLogger(this::class.java)
+
+    @GetMapping("/{language}.json")
+    @PreAuthorize(AUTH_ALL_READ)
+    fun getLocalization(@PathVariable("language") language: String): Any {
+        logger.apiCall("getLocalization", "language" to language)
+        return localizationService.getLocalization(language).let { translation ->
+            ObjectMapper().readValue(translation.localization, Any::class.java)
```

Comment:
Ilmeisesti ilman tätä se tekisi jsonin jonka sisällä on string joka on se koko json? Tuntuu vain aika tarpeettomalta jäsentää tuo jston-string olioksi objectmapperilla jotta jackson voi serialisoida sen takaisin json-stringiksi. Saisikohan ResponseEntityn kautta vain suoraan tuupattua vastauksen bodyksi tuon sisällön...

Reply:
Varsinainen syy tähän oli itse asiassa se, että jos näitä käsitteli puhtaasti stringinä, niin ne palautuivat frontille saakka aina escapetettuina. Nyt aamun tätä katseltuani en oikeastaan vieläkään löytänyt tähän mitään muuta tapaa millä ne saisi palautettua frontille raakana. Tässä lienee joku springin defaultti / konffi joka tuon meillä tekee, mutta mun spring-fu loppuu kesken että tietäisin mihin vipuun pitää koskea